app/lib/router/general_router.dart

```diff
@@ -86,27 +71,17 @@ final generalRoutes = [
   ),
   GoRoute(
     parentNavigatorKey: rootNavKey,
-    name: Routes.linkEmail.name,
-    path: Routes.linkEmail.route,
-    builder: (context, state) => LinkEmailPage(),
-  ),
-  GoRoute(
-    parentNavigatorKey: rootNavKey,
-    name: Routes.uploadAvatar.name,
-    path: Routes.uploadAvatar.route,
-    builder: (context, state) => UploadAvatarPage(),
-  ),
-  GoRoute(
-    parentNavigatorKey: rootNavKey,
-    name: Routes.encryptionBackup.name,
-    path: Routes.encryptionBackup.route,
-    builder: (context, state) => const EncryptionBackupPage(),
-  ),
-  GoRoute(
-    parentNavigatorKey: rootNavKey,
-    name: Routes.redeemInvitations.name,
-    path: Routes.redeemInvitations.route,
-    builder: (context, state) => const RedeemInvitationsPage(),
+    name: Routes.onboarding.name,
+    path: Routes.onboarding.route,
+    builder: (context, state) {
+      final username = state.uri.queryParameters['username'];
+      final isLoginOnboarding =
+          state.uri.queryParameters['isLoginOnboarding'] == 'true';
+      return OnboardingPage(
+        username: username,
+        isLoginOnboarding: isLoginOnboarding,
+      );
+    },
```

Comment:
I think having two onboarding Pages for login and registration really comes in handy here ...